lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -616,3 +616,37 @@ OpFoldResult AsNonPassivePrimOp::fold(ArrayRef<Attribute> operands) {
 
   return {};
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// Connection like things
+///////////////////////////////////////////////////////////////////////////////
+void PartialConnectOp::getCanonicalizationPatterns(
+    OwningRewritePatternList &results, MLIRContext *context) {
+  struct Truncater final : public OpRewritePattern<PartialConnectOp> {
+    using OpRewritePattern::OpRewritePattern;
+
+    LogicalResult matchAndRewrite(PartialConnectOp op,
+                                  PatternRewriter &rewriter) const override {
+      auto destType =
+          op.getOperand(0).getType().cast<FIRRTLType>().getPassiveType();
+      auto srcType = op.getOperand(1).getType().cast<FIRRTLType>();
+      if (destType == srcType)
+        return failure();
+
+      auto srcWidth = srcType.getBitWidthOrSentinel();
+      auto destWidth = destType.getBitWidthOrSentinel();
+
+      if (destType.isa<IntType>() && srcType.isa<IntType>() && srcWidth > 0 &&
+          destWidth > 0 && destWidth < srcWidth) {
+        auto shortened = rewriter.createOrFold<TailPrimOp>(
+            op->getLoc(), destType, op.getOperand(1), srcWidth - destWidth);
+        rewriter.createOrFold<ConnectOp>(op->getLoc(), op.getOperand(0),
+                                         shortened);
+        rewriter.eraseOp(op);
```

Comment:
I think you can use `rewriter.replaceOpWithOp<ConnectOp>(op.getOperand(0), shortened);` which is terser and might be more efficient.